AsyncDisplayKit/ASDisplayNodeExtras.mm

```diff
@@ -10,6 +10,18 @@
 #import "ASDisplayNodeInternal.h"
 #import "ASDisplayNode+FrameworkPrivate.h"
 
+extern ASInterfaceState ASInterfaceStateForDisplayNode(ASDisplayNode *displayNode, UIWindow *window)
+{
+    if (displayNode && [displayNode supportsRangeManagedInterfaceState]) {
+        // Only use the interfaceState of nodes that are range managed
+        ASInterfaceState interfaceState = displayNode.interfaceState;
+        return (window == nil ? (interfaceState &= (~ASInterfaceStateVisible)) : interfaceState);
```

Comment:
// Directly clear the visible bit if we are not in a window.  This means that the interface state is, if not already, about to be set to invisible as it is not possible for an element to be visible while outside of a window.